daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala

```diff
@@ -922,10 +923,11 @@ private[lf] object SBuiltin {
         case v => crash(s"expect optional parties, got: $v")
       }
       val sigs = extractParties(args.get(3))
-      val obs = extractParties(args.get(4))
+      val contractObs = extractParties(args.get(4))
```

Comment:
Given that we need to change it anyway, let's spell it out. I also think speaking of "contract observers" and "choice observers" is somewhat inconsistent. It should either be "template observers" and "choice observers" or "contract observers" and "exercise observers". I heavily prefer the former.
```suggestion
      val templateObservers = extractParties(args.get(4))
```